include/z64collision_check.h

```diff
@@ -9,245 +9,391 @@
     /* 0x00 */ Vec3f quad[4];
 } ColliderQuadDimInit; // size = 0x30
 
-typedef struct {
+typedef struct ColliderQuadElement {
+    /* 0x00 */ ColliderInfo info;
+    /* 0x24 */ ColliderQuadDim dim;
+} ColliderQuadElement; // size = 0x68
+
+typedef struct ColliderQuad {
     /* 0x00 */ Collider base;
-    /* 0x18 */ ColliderBody body;
+    /* 0x18 */ ColliderInfo info;
     /* 0x40 */ ColliderQuadDim dim;
 } ColliderQuad; // size = 0x80
 
-typedef struct {
+typedef struct ColliderQuadInit {
     /* 0x00 */ ColliderInit base;
-    /* 0x08 */ ColliderBodyInit body;
+    /* 0x08 */ ColliderInfoInit info;
     /* 0x20 */ ColliderQuadDimInit dim;
 } ColliderQuadInit; // size = 0x50
 
-typedef struct {
-    /* 0x00 */ ColliderInit_Set3 base;
-    /* 0x08 */ ColliderBodyInit body;
+typedef struct ColliderQuadInitType1 {
+    /* 0x00 */ ColliderInitType1 base;
+    /* 0x08 */ ColliderInfoInit info;
     /* 0x20 */ ColliderQuadDimInit dim;
-} ColliderQuadInit_Set3; // size = 0x50
+} ColliderQuadInitType1; // size = 0x50
 
-typedef struct {
+typedef struct OcLine {
     /* 0x00 */ Linef line;
-    /* 0x18 */ u16 unk_18;
+    /* 0x18 */ u16 ocFlags;
 } OcLine; // size = 0x1C
 
+typedef enum ColliderType {
+    COLTYPE_HIT0, // Blue blood, white hitmark
+    COLTYPE_HIT1, // No blood, dust hitmark
+    COLTYPE_HIT2, // Green blood, dust hitmark
+    COLTYPE_HIT3, // No blood, white hitmark
+    COLTYPE_HIT4, // Water burst, no hitmark
+    COLTYPE_HIT5, // No blood, red hitmark
+    COLTYPE_HIT6, // Green blood, white hitmark
+    COLTYPE_HIT7, // Red blood, white hitmark
+    COLTYPE_HIT8, // Blue blood, red hitmark
+    COLTYPE_METAL,
+    COLTYPE_NONE,
+    COLTYPE_WOOD,
+    COLTYPE_HARD,
+    COLTYPE_TREE
+} ColliderType;
+
+typedef enum ColliderShape {
+    COLSHAPE_JNTSPH,
+    COLSHAPE_CYLINDER,
+    COLSHAPE_TRIS,
+    COLSHAPE_QUAD,
+    COLSHAPE_INVALID
+} ColliderShape;
+
+/* Affects the sound Link's sword makes when hitting it, hookability,
+ * and possibly other things. It's definitely not flags, as all checks
+ * are == or !=. Will probably need more actors decomped to truly 
+ * understand what this is.
+ */
+typedef enum ElementType {
+    ELEMTYPE_UNK0,
+    ELEMTYPE_UNK1,
+    ELEMTYPE_UNK2,
+    ELEMTYPE_UNK3,
+    ELEMTYPE_UNK4,
+    ELEMTYPE_UNK5,
+    ELEMTYPE_UNK6,
+    ELEMTYPE_UNK7
+} ElementType;
+
+
+#define AT_OFF 0 // Cannot have AT collisions when set as AT
+#define AT_ON (1 << 0) // Can have AT collisions when set as AT
+#define AT_HIT (1 << 1) // Had an AT collision
+#define AT_BOUNCED (1 << 2) // Had an AT collision with an AC_HARD collider
+#define AT_PLAYER (1 << 3) // Has player-aligned damage
+#define AT_ENEMY (1 << 4) // Has enemy-aligned damage
+#define AT_OTHER (1 << 5) // Has non-aligned damage
+#define AT_SELF (1 << 6) // Can have AT collisions with colliders attached to the same actor
+#define AT_UNK7 (1 << 7) // Apparently unused
+#define AT_ALL (AT_PLAYER | AT_ENEMY | AT_OTHER) // Has all three damage alignments
```

Comment:
`AT_ALL` (and its `AC`/`OC` equivalents) sounds confusing since it's only certain flags. When seeing this name, I would personally expect it to be the equivalent of every flag toggled.

Maybe we can prefix the 3 relevant flags and the combined flag with the same thing? Perhaps something like `AT_TYPE_PLAYER` and such, with the combined flag as `AT_TYPE_ALL`?

Reply:
`TYPE` prefix should work.

Reply:
Although `OC_TYPE_PLAYER` is what `OT_PLAYER` is short for.